main.go

```diff
@@ -428,6 +406,113 @@ func main() {
+	log.Infof("Sign Build Artifact with APKSigner: %s", unsignedBuildArtifactPth)
+
+	alignedPath, err := zipAlignArtifact(zipalign, unsignedBuildArtifactPth, buildArtifactDir, buildArtifactBasename, artifactExt, "aligned", "", pageAlignConfig)
+
+	if err != nil {
+		failf("Failed to zip align artifact, error: %s", err)
+	}
+
+	signedArtifactName := fmt.Sprintf("%s-bitrise-signed%s", buildArtifactBasename, artifactExt)
+	if artifactName := fmt.Sprintf("%s%s", outputName, artifactExt); outputName != "" {
+		log.Printf("- Exporting (%s) as: %s", signedArtifactName, artifactName)
+		signedArtifactName = artifactName
+	}
+	fullPath := filepath.Join(buildArtifactDir, signedArtifactName)
+
+	log.Infof("Signing build: %s", alignedPath)
+
+	err = apkSigner.SignBuidlArtifact(alignedPath, fullPath)
+	if err != nil {
+		failf("Failed to build artifact, error: %s", err)
+	}
+
+	log.Infof("Verify Build Artifact")
+	err = apkSigner.VerifyBuildArtifact(fullPath)
+	if err != nil {
+		failf("Failed to build artifact, error: %s", err)
+	}
+
+	return fullPath
+}
+
+func prepareBuildArtifact(buildArtifactPath string, unsignedBuildArtifactPth string, aapt string) {
+	if err := command.CopyFile(buildArtifactPath, unsignedBuildArtifactPth); err != nil {
+		failf("Failed to copy build artifact, error: %s", err)
+	}
+
+	isSigned, err := isBuildArtifactSigned(aapt, unsignedBuildArtifactPth)
+	if err != nil {
+		failf("Failed to check if build artifact is signed, error: %s", err)
+	}
+
+	if isSigned {
+		log.Printf("Signature file (DSA or RSA) found in META-INF, unsigning the build artifact...")
+		if err := unsignBuildArtifact(aapt, unsignedBuildArtifactPth); err != nil {
+			failf("Failed to unsign Build Artifact, error: %s", err)
+		}
+		fmt.Println()
+	} else {
+		log.Printf("No signature file (DSA or RSA) found in META-INF, skipping build artifact unsign...")
+		fmt.Println()
+	}
+	// ---
+}
+
+func zipAlignArtifact(zipalign, unalignedBuildArtifactPth string, buildArtifactDir string, buildArtifactBasename string, artifactExt string, fullPathExt string, outputName string, pageAlignConfig pageAlignStatus) (string, error) {
+	log.Infof("Zipalign Build Artifact")
+	signedArtifactName := fmt.Sprintf("%s-bitrise-%s%s", buildArtifactBasename, fullPathExt, artifactExt)
+	if artifactName := fmt.Sprintf("%s%s", outputName, artifactExt); outputName != "" {
+		log.Printf("- Exporting (%s) as: %s", signedArtifactName, artifactName)
+		signedArtifactName = artifactName
+	}
+	fullPath := filepath.Join(buildArtifactDir, signedArtifactName)
+
+	pageAlign := pageAlignConfig == pageAlignYes
+	// Only care about .so memory page alignment for APKs
+	if !strings.EqualFold(artifactExt, ".aab") && pageAlignConfig == pageAlignAuto {
+		extractNativeLibs, err := parseAPKextractNativeLibs(fullPath)
```

Comment:
`unalignedBuildArtifactPth` should be used, as there is no file at `fullPath` at this point.